unit_test/solver/network/assets/test_base_asset.py

```diff
@@ -0,0 +1,302 @@
+
+        # Assert
+        assert result is self.supply_node
+
+    def test_base_is_all_connected(self) -> None:
+        """Test the is_all_connected method."""
+        # Arrange
+        self.asset.connect_node(node=self.supply_node, connection_point=0)
+        self.asset.connect_node(node=self.return_node, connection_point=1)
+
+        # Act
+        result = self.asset.is_all_connected()
+
+        # Assert
+        assert result is True
+
+    def test_base_is_not_all_connected(self) -> None:
+        """Test the is_all_connected method."""
+        # Arrange
+        self.asset.connect_node(node=self.supply_node, connection_point=0)
+
+        # Act
+        result = self.asset.is_all_connected()
+
+        # Assert
+        assert result is False
+
+    def test_base_add_thermal_equations_with_discharge(self) -> None:
+        """Test the add_thermal_equations method.
+
+        The discharge at the connection point of the prev_sol is greater than 0.
+        """
+        # Arrange
+        connection_point_id = 0
+        self.asset.connect_node(node=self.supply_node, connection_point=connection_point_id)
+        self.asset.prev_sol[IndexEnum.discharge + connection_point_id * NUMBER_CORE_QUANTITIES] = (
+            1.0
+        )
+
+        # Act
+        equation_object = self.asset.add_thermal_equations(connection_point=connection_point_id)
+
+        # Assert
+        assert equation_object.rhs == fluid_props.get_ie(self.asset.supply_temperature)
+        assert all(equation_object.coefficients == [1.0])
+        assert all(
+            equation_object.indices
+            == [IndexEnum.internal_energy + connection_point_id * NUMBER_CORE_QUANTITIES]
+        )
+
+    def test_base_add_thermal_equations_no_discharge(self) -> None:
+        """Test the add_thermal_equations method.
+
+        The discharge at the connection point of the prev_sol is 0.
+        """
+        # Arrange
+        connection_point_id = 0
+        self.asset.connect_node(node=self.supply_node, connection_point=connection_point_id)
+
+        # Act
+        equation_object = self.asset.add_thermal_equations(connection_point=connection_point_id)
+
+        # Assert
+        assert equation_object.rhs == 0.0
+        assert all(equation_object.coefficients == [1.0, -1.0])
+        assert all(
+            equation_object.indices == [IndexEnum.internal_energy, IndexEnum.internal_energy]
+        )
+
+    def test_base_add_thermal_equations_no_discharge_not_connected(self) -> None:
+        """Test the add_thermal_equations method.
+
+        The discharge at the connection point of the prev_sol is 0.
+        """
+        # Arrange
+        connection_point_id = 0
+
+        # Act
+        with self.assertRaises(ValueError) as cm:
+            self.asset.add_thermal_equations(connection_point=connection_point_id)
+
+        # Assert
+        self.assertIsInstance(cm.exception, ValueError)
+        self.assertEqual(
```

Comment:
hier gebruik je andere assert dan verderop. We zouden toch met deze verder gaan?

Reply:
#75